x/axelarnet/module.go

```diff
@@ -187,105 +170,68 @@ func (am AppModule) BeginBlock(ctx sdk.Context, req abci.RequestBeginBlock) {
 // EndBlock executes all state transitions this module requires at the end of each new block
 func (am AppModule) EndBlock(ctx sdk.Context, req abci.RequestEndBlock) []abci.ValidatorUpdate {
 	return utils.RunCached(ctx, am.keeper, func(ctx sdk.Context) ([]abci.ValidatorUpdate, error) {
-		return EndBlocker(ctx, req, am.keeper, am.ibcK)
+		return EndBlocker(ctx, req, am.keeper, am.keeper)
 	})
 }
 
 // ConsensusVersion implements AppModule/ConsensusVersion.
 func (AppModule) ConsensusVersion() uint64 { return 6 }
 
-// OnChanOpenInit implements the IBCModule interface
-func (am AppModule) OnChanOpenInit(
-	ctx sdk.Context,
-	order channeltypes.Order,
-	connectionHops []string,
-	portID string,
-	channelID string,
-	chanCap *capabilitytypes.Capability,
-	counterparty channeltypes.Counterparty,
-	version string,
-) (string, error) {
-	return am.transferModule.OnChanOpenInit(ctx, order, connectionHops, portID, channelID, chanCap, counterparty, version)
-}
-
-// OnChanOpenTry implements the IBCModule interface
-func (am AppModule) OnChanOpenTry(
-	ctx sdk.Context,
-	order channeltypes.Order,
-	connectionHops []string,
-	portID,
-	channelID string,
-	chanCap *capabilitytypes.Capability,
-	counterparty channeltypes.Counterparty,
-	counterpartyVersion string,
-) (string, error) {
-	return am.transferModule.OnChanOpenTry(ctx, order, connectionHops, portID, channelID, chanCap, counterparty, counterpartyVersion)
-}
-
-// OnChanOpenAck implements the IBCModule interface
-func (am AppModule) OnChanOpenAck(
-	ctx sdk.Context,
-	portID,
-	channelID string,
-	counterpartyChannelID string,
-	counterpartyVersion string,
-) error {
-	return am.transferModule.OnChanOpenAck(ctx, portID, channelID, counterpartyChannelID, counterpartyVersion)
-}
-
-// OnChanOpenConfirm implements the IBCModule interface
-func (am AppModule) OnChanOpenConfirm(
-	ctx sdk.Context,
-	portID,
-	channelID string,
-) error {
-	return am.transferModule.OnChanOpenConfirm(ctx, portID, channelID)
-}
-
-// OnChanCloseInit implements the IBCModule interface
-func (am AppModule) OnChanCloseInit(
-	ctx sdk.Context,
-	portID,
-	channelID string,
-) error {
-	return am.transferModule.OnChanCloseInit(ctx, portID, channelID)
+// AxelarnetIBCModule is an IBCModule that adds rate limiting and gmp processing to the ibc middleware
+type AxelarnetIBCModule struct {
+	porttypes.IBCModule
+	keeper      keeper.Keeper
+	nexus       types.Nexus
+	ibcK        keeper.IBCKeeper
```

Comment:
It's confusing to me that we're now extracting the base keeper from `ibcK` for `AppModule` but keeping them separate for this struct.